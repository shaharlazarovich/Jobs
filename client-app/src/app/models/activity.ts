export interface IAttendee {
    userName: string;
    displayName: string;
    image: string;
    isHost: boolean;
    following?: boolean,
}

//we are adding this interface to mimic what we do on the server for paging -
//which is to have an entity around our activity array, in order to add a count
//of activities - which then, we could device in the number of pages
export interface IActivitiesEnvelope {
    activities: IActivity[];
    activityCount: number
}

export interface IActivity {
    id: string;
    title: string;
    description: string;
    category: string;
    date: Date;
    city: string;
    venue: string;
    isGoing: boolean;
    isHost: boolean;
    attendees: IAttendee[];
    comments: IComment[];
}

export interface IComment {
    id: string;
    createdAt: Date;
    body: string;
    username: string;
    displayName: string;
    image: string;
}

export interface IActivityFormValues extends Partial<IActivity> {
    time?: Date    
}

export class ActivityFormValues implements IActivityFormValues {
    id?: string = undefined;
    title: string = '';
    category: string = '';
    description: string = '';
    date?:Date = undefined;
    time?: Date = undefined;
    city: string = '';
    venue: string = '';

    constructor(init?: IActivityFormValues) {
        if (init && init.date) {
            init.time = init.date;
        }
        Object.assign(this, init); //this will initialize our object of class ActivityFormValues
                                  // with the above values
    }
}