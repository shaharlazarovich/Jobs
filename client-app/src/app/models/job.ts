//we are adding this interface to mimic what we do on the server for paging -
//which is to have an entity around our jobs array, in order to add a count
//of jobs - which then, we could device in the number of pages
export interface IJobsEnvelope {
    jobs: IJob[];
    jobsCount: number
}

export interface IJob {
    id: number;
    jobName: string;
    company: string;
    replication: string;
    servers: string;
    lastRun: Date;
    date: Date;
    time: Date;
    rta: string;
    results: string;
    key: string;
    rtoNeeded: string;
    jobIP: string;
}


export interface IJobFormValues extends Partial<IJob> {
     
}

export class JobFormValues implements IJobFormValues {
    id?: number = 0;
    jobName: string = '';
    company: string = '';
    replication: string = '';
    servers: string = '';
    lastRun?:Date = undefined;
    date?:Date = undefined;
    time?:Date = undefined;
    rta?: string = '';
    results: string = '';
    key: string = '';
    rtoNeeded: string = '';
    jobIP: string = '';

    constructor(init?: IJobFormValues) {
        Object.assign(this, init); //this will initialize our object of class JobFormValues
                                  // with the above values
    }
}