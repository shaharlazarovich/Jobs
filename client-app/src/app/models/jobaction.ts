//we are adding this interface to mimic what we do on the server for paging -
//which is to have an entity around our jobs array, in order to add a count
//of jobs - which then, we could device in the number of pages
export interface IJobActionsEnvelope {
    jobActions: IJobAction[];
    jobActionsCount: number
}

export interface IJobAction {
    id: string;
    jobId: string;
    jobName: string;
    userId: string;
    actionDate: Date;
    remoteIP: string;
    remoteResponse: string;
    requestProperties: string;
    source: string;
    action: string;
}

export class JobAction implements IJobAction {
    id: string = '';
    jobId: string = '';
    jobName: string = '';
    userId: string = '';
    actionDate: Date = new Date();
    remoteIP: string = '';
    remoteResponse: string = '';
    requestProperties: string = '';
    source: string = '';
    action: string = '';

    constructor(init?: IJobAction) {
        Object.assign(this, init); //this will initialize our object of class JobActions
                                  // with the above values
    }
}