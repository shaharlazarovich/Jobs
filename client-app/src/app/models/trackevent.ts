//we are adding this interface to mimic what we do on the server for paging -
//which is to have an entity around our jobs array, in order to add a count
//of jobs - which then, we could device in the number of pages
export interface ITrackEventsEnvelope {
    trackEvents: ITrackEvent[];
    trackEventsCount: number
}

export interface ITrackEvent {
    id: string;
    jobId?: string;
    jobName?: string;
    userId?: string;
    actionDate: Date;
    remoteIP?: string;
    remoteResponse?: string;
    requestProperties?: string;
    source: string;
    event: string;
}

export class TrackEvent implements ITrackEvent {
    id: string = '';
    jobId?: string = '';
    jobName?: string = '';
    userId?: string = '';
    actionDate: Date = new Date();
    remoteIP?: string = '';
    remoteResponse?: string = '';
    requestProperties?: string = '';
    source: string = '';
    event: string = '';

    constructor(init?: ITrackEvent) {
        Object.assign(this, init); //this will initialize our object of class JobActions
                                  // with the above values
    }
}