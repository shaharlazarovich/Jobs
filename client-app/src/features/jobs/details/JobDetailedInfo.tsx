import React from 'react'
import { Segment, Grid, Icon } from 'semantic-ui-react';
import { IJob } from '../../../app/models/job';
import {format} from 'date-fns'


const JobDetailedInfo: React.FC<{job: IJob}> = ({job}) => {
    return (
        <Segment.Group data-test="component-job-detailed-info">
              <Segment attached='top'>
                <Grid>
                  <Grid.Column width={1}>
                    <Icon size='large' color='teal' name='info' />
                  </Grid.Column>
                  <Grid.Column width={15}>
                    <p>{job.company}</p>
                  </Grid.Column>
                </Grid>
              </Segment>
              <Segment attached>
                <Grid verticalAlign='middle'>
                  <Grid.Column width={1}>
                    <Icon name='calendar' size='large' color='teal' />
                  </Grid.Column>
                  <Grid.Column width={15}>
                    <span>
                      {format(job.lastRun,'eeee do MM')} at {format(job.lastRun,'h:mm')}
                    </span>
                  </Grid.Column>
                </Grid>
              </Segment>
              <Segment attached>
                <Grid verticalAlign='middle'>
                  <Grid.Column width={1}>
                    <Icon name='marker' size='large' color='teal' />
                  </Grid.Column>
                  <Grid.Column width={11}>
                    <span>{job.rtoNeeded}, {job.rta}</span>
                  </Grid.Column>
                </Grid>
              </Segment>
            </Segment.Group>
    )
};

export default JobDetailedInfo
