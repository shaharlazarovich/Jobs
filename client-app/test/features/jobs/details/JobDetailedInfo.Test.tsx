import { ReactWrapper } from "enzyme";
import React from "react";
import { IJob } from '../../../../src/app/models/job'
import { findByTestAttr } from '../../../common/testUtils';
import JobDetailedInfo from "../../../../src/features/jobs/details/JobDetailedInfo";
import { jobMock } from "../../../common/mocks";
import { setupWrapper } from "../../../common/wrapper";

describe('Job Detailed Info UniTest', () => {
    let wrapper: ReactWrapper;
    let testJob: IJob;
    beforeEach(async () => {
        testJob = { ...jobMock };
        wrapper = await setupWrapper(<JobDetailedInfo job={testJob} />);
    });
    test('renders without error', () => {
        const component = findByTestAttr(wrapper, 'component-job-detailed-info');
        expect(component.length).toBe(1);
    });
    test('should match the snapshot', () => {
        const component = findByTestAttr(wrapper, 'component-job-detailed-info');
        expect(component).toMatchSnapshot();
    });
})