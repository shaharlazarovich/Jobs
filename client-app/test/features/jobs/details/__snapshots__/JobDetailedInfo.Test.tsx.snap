// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Job Detailed Info UniTest should match the snapshot 1`] = `
<SegmentGroup
  data-test="component-job-detailed-info"
>
  <div
    className="ui segments"
    data-test="component-job-detailed-info"
  >
    <Segment
      attached="top"
    >
      <div
        className="ui top attached segment"
      >
        <Grid>
          <div
            className="ui grid"
          >
            <GridColumn
              width={1}
            >
              <div
                className="one wide column"
              >
                <Icon
                  as="i"
                  color="teal"
                  name="info"
                  size="large"
                >
                  <i
                    aria-hidden="true"
                    className="teal info large icon"
                    onClick={[Function]}
                  />
                </Icon>
              </div>
            </GridColumn>
            <GridColumn
              width={15}
            >
              <div
                className="fifteen wide column"
              >
                <p>
                  Shomera
                </p>
              </div>
            </GridColumn>
          </div>
        </Grid>
      </div>
    </Segment>
    <Segment
      attached={true}
    >
      <div
        className="ui attached segment"
      >
        <Grid
          verticalAlign="middle"
        >
          <div
            className="ui middle aligned grid"
          >
            <GridColumn
              width={1}
            >
              <div
                className="one wide column"
              >
                <Icon
                  as="i"
                  color="teal"
                  name="calendar"
                  size="large"
                >
                  <i
                    aria-hidden="true"
                    className="teal calendar large icon"
                    onClick={[Function]}
                  />
                </Icon>
              </div>
            </GridColumn>
            <GridColumn
              width={15}
            >
              <div
                className="fifteen wide column"
              >
                <span>
                  Monday 6th 01
                   at 
                  11:00
                </span>
              </div>
            </GridColumn>
          </div>
        </Grid>
      </div>
    </Segment>
    <Segment
      attached={true}
    >
      <div
        className="ui attached segment"
      >
        <Grid
          verticalAlign="middle"
        >
          <div
            className="ui middle aligned grid"
          >
            <GridColumn
              width={1}
            >
              <div
                className="one wide column"
              >
                <Icon
                  as="i"
                  color="teal"
                  name="marker"
                  size="large"
                >
                  <i
                    aria-hidden="true"
                    className="teal marker large icon"
                    onClick={[Function]}
                  />
                </Icon>
              </div>
            </GridColumn>
            <GridColumn
              width={11}
            >
              <div
                className="eleven wide column"
              >
                <span>
                  10
                  , 
                  10
                </span>
              </div>
            </GridColumn>
          </div>
        </Grid>
      </div>
    </Segment>
  </div>
</SegmentGroup>
`;
